### PR TITLE
LibJS: Array.prototype.slice

### PR DESCRIPTION
**Pull request by kessejones**
Added slice function in Array.prototype.

### CODE REVIEW

File: Libraries/LibJS/Runtime/ArrayPrototype.cpp [hunk 1]
```diff
@@ -254,4 +255,42 @@ Value ArrayPrototype::concat(Interpreter& interpreter)
...
+Value ArrayPrototype::slice(Interpreter& interpreter)
+{
+    auto* array = array_from(interpreter);
+    if (!array)
+        return {};
+
+    auto* new_array = Array::create(interpreter.global_object());
+    if (interpreter.argument_count() == 0) {
+        new_array->elements().append(array->elements());
+    } else {
```
Comment (awesomekling) [inline Libraries/LibJS/Runtime/ArrayPrototype.cpp:267]: Let's just return early here, then we don't need `else` and can avoid nesting.

File: Libraries/LibJS/Runtime/ArrayPrototype.cpp [hunk 2]
```diff
@@ -254,4 +255,42 @@ Value ArrayPrototype::concat(Interpreter& interpreter)
...
+        ssize_t array_size = static_cast<ssize_t>(array->elements().size());
+        auto start_slice = interpreter.argument(0).to_i32();
+        auto end_slice = array_size;
+
+        if (start_slice > array_size)
+            return Value(new_array);
+        else if (start_slice < 0)
```
Comment (awesomekling) [inline Libraries/LibJS/Runtime/ArrayPrototype.cpp:274]: No need for `else` after `return`.

File: Libraries/LibJS/Runtime/ArrayPrototype.cpp [hunk 3]
```diff
@@ -254,4 +255,42 @@ Value ArrayPrototype::concat(Interpreter& interpreter)
...
+    auto* new_array = Array::create(interpreter.global_object());
+    if (interpreter.argument_count() == 0) {
+        new_array->elements().append(array->elements());
+    } else {
+        ssize_t array_size = static_cast<ssize_t>(array->elements().size());
+        auto start_slice = interpreter.argument(0).to_i32();
+        auto end_slice = array_size;
+
+        if (start_slice > array_size)
+            return Value(new_array);
```
Comment (awesomekling) [inline Libraries/LibJS/Runtime/ArrayPrototype.cpp:273]: Object pointers like `Array*` can be returned directly:
```cpp
return new_array;
```

File: Libraries/LibJS/Tests/Array.prototype.slice.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+load("test-common.js");
+
+try {
+    assert(Array.prototype.concat.length === 1);
```
Comment (linusg) [inline Libraries/LibJS/Tests/Array.prototype.slice.js:4]:
```suggestion
    assert(Array.prototype.slice.length === 1);
```

Comment (kessejones) [reply Libraries/LibJS/Tests/Array.prototype.slice.js:4]: thanks, it was because of ctrl+c and ctrl+v. :)

**Comment by awesomekling** [comment]
👍

**Pull request merged**